File: examples/async/gateio_ws_public.py
"""Subscribe to a Gate.io public WebSocket stream."""

import asyncio

from dcex.ws import gateio


async def main() -> None:
    """Subscribe to Gate.io spot trades and print the subscription response."""
    async with gateio.public() as ws:
        await ws.subscribe_trades("BTC-USDT-SPOT")
        print(await ws.recv())


if __name__ == "__main__":
    asyncio.run(main())

File: crates/dcex/examples/gateio_ws_public.rs
use std::time::Duration;

use dcex::ws::gateio::GateioPublicWebSocket;

#[tokio::main]
async fn main() -> dcex::Result<()> {
    let mut ws = GateioPublicWebSocket::new(Duration::from_secs(10))?;
    ws.connect().await?;
    ws.subscribe_trades("BTC-USDT-SPOT").await?;
    println!("{}", ws.recv().await?);
    println!("{}", ws.recv().await?);
    ws.close().await?;
    Ok(())
}
